hera_cal/utils.py

```diff
@@ -310,6 +312,128 @@ def JD2RA(JD, longitude=21.42830, latitude=-30.72152, epoch='current'):
         return RA[0]
 
 
+def get_sun_alt(jds, longitude=21.42830, latitude=-30.72152):
+    """
+    Given longitude and latitude, get the Solar alittude at a given time.
+
+    Parameters
+    ----------
+    jds : float or ndarray of floats
+        Array of Julian Dates
+
+    longitude : float
+        Longitude of observer in degrees East
+
+    latitude : float
+        Latitude of observer in degrees North
+
+    Returns
+    -------
+    alts : float or ndarray
+        Array of altitudes [degrees] of the Sun
+    """
+    # type check
+    array = True
+    if isinstance(jds, (float, np.float, np.float64, int, np.int, np.int32)):
+        jds = [jds]
+        array = False 
+
+    # get earth location
+    e = crd.EarthLocation(lat=latitude*unt.deg, lon=longitude*unt.deg)
+
+    # get AltAz frame
+    a = crd.AltAz(location=e)
+
+    # get Sun locations
+    alts = np.array(map(lambda t: crd.get_sun(Time(t, format='jd')).transform_to(a).alt.value, jds))
+
+    if array:
+        return alts
+    else:
+        return alts[0]
+
+ 
+def solar_flag(flags, start_jd, time_array=None, flag_alt=0.0, longitude=21.42830, latitude=-30.72152, 
```

Comment:
I don't see what's 1) complicated about it and 2) what isn't doing much: there is no repetition or redundancy, so I don't think it could be expressed in fewer lines and also I think it's perfectly reasonable to have a function that flags entire datacontainers as something an analyst might want handy... what's your argument for why we should *decrease* modularity on this front?

Reply:
or, put another way, what's your few line solution?

Reply:
Let me take a stab at it.

Reply:
Thoughts? It passes your unit tests.